app/views/contexts/calculated_summary_context.py

```diff
@@ -30,78 +35,112 @@ def build_groups_for_section(self, section, return_to_block_id):
                 self._metadata,
                 self._response_metadata,
                 self._schema,
-                location,
+                current_location,
                 self._language,
                 return_to="calculated-summary",
                 return_to_block_id=return_to_block_id,
             ).serialize()
             for group in section["groups"]
         ]
 
-    def build_view_context_for_calculated_summary(self, current_location):
-        block = self._schema.get_block(current_location.block_id)
+    def build_view_context_for_calculated_summary(
+        self, current_location: Location
+    ) -> Mapping:
+        if (block_id := current_location.block_id) and (
+            block := self._schema.get_block(block_id)
+        ):
```

Comment:
If this was done just for type hinting I suggest removing it, we are making the code less readable for type hints. You can override it as it shouldn't get this far if the block was invalid.

Reply:
Updated now 👍

Reply:
Still outstanding.